be/src/exec/pipeline/hashjoin/hash_join_build_operator.cpp

```diff
@@ -25,21 +33,23 @@ void HashJoinBuildOperator::finish(RuntimeState* state) {
     }
 }
 
-HashJoinBuildOperatorFactory::HashJoinBuildOperatorFactory(int32_t id, int32_t plan_node_id, HashJoiner* hash_joiner)
-        : OperatorFactory(id, "hash_join_build", plan_node_id), _hash_joiner(hash_joiner) {}
+HashJoinBuildOperatorFactory::HashJoinBuildOperatorFactory(int32_t id, int32_t plan_node_id,
+                                                           HashJoinerFactoryPtr hash_joiner_factory)
+        : OperatorFactory(id, "hash_join_build", plan_node_id), _hash_joiner_factory(hash_joiner_factory) {}
 
 Status HashJoinBuildOperatorFactory::prepare(RuntimeState* state) {
     RETURN_IF_ERROR(OperatorFactory::prepare(state));
-    return _hash_joiner->prepare(state);
+    return _hash_joiner_factory->prepare(state);
 }
 
 void HashJoinBuildOperatorFactory::close(RuntimeState* state) {
-    _hash_joiner->close(state);
+    _hash_joiner_factory->close(state);
```

Comment:
RETURN_IF_ERROR

Reply:
the return type of OperatorFactory::close is void, too. if returning-void is considered to be improper, another dedicated PR can be used to resolve this problem.